### PR TITLE
Add new changes from Ft add remove to Development brach

### PR DESCRIPTION
**Pull request by aristides1000**
The following changes were made:

- Remove all hardcoded items from the tasks array.
- Create a new JavaScript file for the new functionality.
- Implement a function for adding a new task (add a new element to the array).
- Implement a function for editing task descriptions.
- Implement a function for deleting a task (remove an element from the array).
- Implement a function for the "Clear all completed" button (use filter() method).
- By default new tasks should have the property completed set to false and the property index set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5).
- Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.
- All changes to the To Do List should be saved in local storage.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -26,26 +27,83 @@ function displayTasks() {
...
 
+const addNewDescription = document.getElementById('addNewDescription');
+const inputNewDescription = document.getElementById('inputNewDescription');
+
+addNewDescription.addEventListener('click', () => {
+  let activities = loadDataLocalStorage();
+  activities = addTask(activities, inputNewDescription.value);
+  saveDataLocalStorage(activities);
+  displayTasks();
+});
```
Comment (Lameck1) [inline src/index.js]: - [OPTIONAL] Right now it's possible to add empty tasks. You can run a check on `inputNewDescription.value` to ascertain its truthiness. If no value, do not add a task, if value, add a task. You can go as far as preventing the addition of duplicate tasks.

File: src/add-and-remove.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+function deleteTask(activities, index) {
+  activities.splice(index, 1);
+  return activities;
+}
+
+function deleteAll(activities) {
+  const result = activities.filter((activity) => (activity.completed === false));
+  return result;
+}
```
Comment (Lameck1) [inline src/add-and-remove.js]: - [x] One requirement has not been met here. Look at the quote below 👇🏼:

> Deleting a task should update all remaining items' indexes, so they represent the current list order and are unique.

Take a look at the following screenshots in order:

| **Added some tasks** |
| --- |
| one |

| **After deleting the `first task` and the `third task`** |
| --- |
| three |

Looking at the screenshots above, the final indexes should have been 0 and 1 respectively after deleting some tasks basing on where your numbering has started(ideally, 1 and 2 according to the requirement. Let your indexing start from 1). **_DON'T look at the index references from the localStorage Array, those are the obvious default ones and are not under test in this task._** So, there needs an implementation of a function that will update the tasks indexes after a delete operation. To further break it down, deleting a task with an index value of 1 should make the next task's index be updated to 1 and the one after it 2, so on so forth. The order of 1,2,3... must remain true whatever happens to the data. Please work on this.

**Review by Lameck1 — CHANGES_REQUESTED** [review]
Hi @aristides1000, 

Good job so far!

### Highlights
* Proffessional README ✔️ 
* Commit messages ✔️ (We can do better by making them precise. Utilise the option of adding description if maybe you want to be more elaborate with the commit message). Go through this [GUIDE](https://www.freecodecamp.org/news/writing-good-commit-messages-a-practical-guide/) to more insights.
* Using JS events ✔️
* Using localStorage (CRUD operations) ✔️

We still however have a minor issue that we need to address for us to move ahead.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by hemantso — APPROVED** [review]
Hi, @aristides1000, 

Good job on making the requested changes. 👏 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

## Congratulations! 🎉 

Alt Text

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- It will be better if we make the input field empty after adding a task. Kindly try to fix it.

Cheers and happy coding 😃

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. You can mention me (@hemant-soni-vst-au4) so I can see your comments and respond.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**